### ISSUE TITLE
Alertmanager behaviour changed on upgrade from 19.09 -> 20.03

### ISSUE THREAD

**Issue by ptman** [NixOS/nixpkgs#85660]
**Describe the bug**

Prometheus Alertmanager just shows "Loading..." instead of providing the usual interface. Also grafana shows an error trying to fetch info from alertmanager.

**To Reproduce**

I upgraded from 19.09 -> 20.03.

**Expected behavior**

I would've expected alertmanager to continue working. Prometheus seems to work, and is much more complex.

**Additional context**

Found no errors in `journalctl -u alertmanager -e`

**Notify maintainers**

@benley @fpletz @globin 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.33, NixOS, 20.03.1424.e79d1e83f91 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.03.1424.e79d1e83f91"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by globin** [comment]
Hmm, it definitely is still working for us, could you try and get more debug information?

**Comment by ptman** [comment]
It seems that I get alerts via email, just the web interface is somehow stuck. Any ideas on how to get more info?

**Comment by globin** [comment]
You could start it with `--log.level=debug` and look in the browser console for errors.

**Comment by ptman** [comment]
Ok, I made some progress. I did look at the browser console and network requests, but didn't immediately spot anything. Now I noticed that if I access alertmanager via ssh port forwarding ( ssh -L 8080:localhost:9093 ...@alertmanager-nixos ) it works. So the problem is related to the reverse proxy. And I compared the console outputs. It seems that only through the reverse proxy and using HTTPS do I get this error:

    SecurityError: The operation is insecure. (script.js:1)

script.js is of course minified.

And this issue only occurs on Firefox, not chromium.

And this also doesn't fix the issue with grafana. But it seems that alertmanager isn't necessarily at fault
